### ISSUE TITLE
kube-dns keeps crash loop

### ISSUE THREAD

**Issue by zhangqx2010** [kubernetes/kubernetes#48803]
/kind bug

**What happened**:
Using kubelet 1.7, after setting up kube cluster. The kube-dns pod keeps crash loop.

```
[root@m01v ~]# kubectl get nodes
NAME                         STATUS    AGE       VERSION
c01v                           Ready     14m       v1.7.0
c02v                           Ready     14m       v1.7.0
c03v                           Ready     14m       v1.7.0
m01v                          Ready     17m       v1.7.0
m02v                          Ready     14m       v1.7.0
m03v                          Ready     14m       v1.7.0
```

```
[root@m01v ~]# kubectl get svc --all-namespaces 
NAMESPACE     NAME          CLUSTER-IP      EXTERNAL-IP   PORT(S)         AGE
default       kubernetes    10.96.0.1       <none>        443/TCP         12m
kube-system   calico-etcd   10.96.232.136   <none>        6666/TCP        3m
kube-system   kube-dns      10.96.0.10      <none>        53/UDP,53/TCP   12m
```

```
[root@m01v ~]# kubectl get pod -n kube-system 
NAME                                                 READY     STATUS             RESTARTS   AGE
calico-etcd-3x5f5                                    1/1       Running            0          16m
calico-etcd-mlphz                                    1/1       Running            0          16m
calico-etcd-vvhj5                                    1/1       Running            0          16m
calico-node-4321c                                    2/2       Running            0          15m
calico-node-6jghk                                    2/2       Running            0          15m
calico-node-8zrw3                                    2/2       Running            0          16m
calico-node-9krn6                                    2/2       Running            0          15m
calico-node-dzwgg                                    2/2       Running            0          15m
calico-node-jr70t                                    2/2       Running            0          15m
calico-policy-controller-1727037546-5dr85            1/1       Running            0          15m
etcd-m01v                                            1/1       Running            0          23m
kube-apiserver-m01v                                  1/1       Running            0          23m
kube-controller-manager-m01v                         1/1       Running            0          23m
kube-dns-2425271678-l66cq                            0/3       CrashLoopBackOff   18         24m
kube-proxy-01xkw                                     1/1       Running            0          21m
kube-proxy-0hd7r                                     1/1       Running            0          21m
kube-proxy-3b6xm                                     1/1       Running            0          21m
kube-proxy-544mp                                     1/1       Running            0          21m
kube-proxy-p80c1                                     1/1       Running            0          24m
kube-proxy-vn961                                     1/1       Running            0          21m
kube-scheduler-m01v                                  1/1       Running            0          23m
```

```
[root@m01v ~]# kubectl describe pod kube-dns-2425271678-l66cq  -n kube-system 
Name:		kube-dns-2425271678-l66cq
Namespace:	kube-system
Node:		m03v/10.20.101.43
Start Time:	Wed, 12 Jul 2017 17:31:00 +0800
Labels:		k8s-app=kube-dns
		pod-template-hash=2425271678
Annotations:	kubernetes.io/created-by={"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"ReplicaSet","namespace":"kube-system","name":"kube-dns-2425271678","uid":"a770084d-66e3-11e7-900d-00505682...
		scheduler.alpha.kubernetes.io/critical-pod=
Status:		Running
IP:		192.168.54.0
Created By:	ReplicaSet/kube-dns-2425271678
Controlled By:	ReplicaSet/kube-dns-2425271678
Containers:
  kubedns:
    Container ID:	docker://660b32a63da56b44fee2b091c7e71748b37c8f1b51300f7d5a78319a5e304064
    Image:		gcr.io/google_containers/k8s-dns-kube-dns-amd64:1.14.4
    Image ID:		docker-pullable://gcr.io/google_containers/k8s-dns-kube-dns-amd64@sha256:40790881bbe9ef4ae4ff7fe8b892498eecb7fe6dcc22661402f271e03f7de344
    Ports:		10053/UDP, 10053/TCP, 10055/TCP
    Args:
      --domain=cluster.local.
      --dns-port=10053
      --config-dir=/kube-dns-config
      --v=2
    State:		Waiting
      Reason:		CrashLoopBackOff
    Last State:		Terminated
      Reason:		Error
      Exit Code:	255
      Started:		Wed, 12 Jul 2017 17:34:58 +0800
      Finished:		Wed, 12 Jul 2017 17:35:58 +0800
    Ready:		False
    Restart Count:	3
    Limits:
      memory:	170Mi
    Requests:
      cpu:	100m
      memory:	70Mi
    Liveness:	http-get http://:10054/healthcheck/kubedns delay=60s timeout=5s period=10s #success=1 #failure=5
    Readiness:	http-get http://:8081/readiness delay=3s timeout=5s period=10s #success=1 #failure=3
    Environment:
      PROMETHEUS_PORT:	10055
    Mounts:
      /kube-dns-config from kube-dns-config (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-dns-token-xhmbk (ro)
  dnsmasq:
    Container ID:	docker://83c6f750c4dee1777c9056006c5dbe09d656db9b74d5f472efdea877e2613d69
    Image:		gcr.io/google_containers/k8s-dns-dnsmasq-nanny-amd64:1.14.4
    Image ID:		docker-pullable://gcr.io/google_containers/k8s-dns-dnsmasq-nanny-amd64@sha256:aeeb994acbc505eabc7415187cd9edb38cbb5364dc1c2fc748154576464b3dc2
    Ports:		53/UDP, 53/TCP
    Args:
      -v=2
      -logtostderr
      -configDir=/etc/k8s/dns/dnsmasq-nanny
      -restartDnsmasq=true
      --
      -k
      --cache-size=1000
      --log-facility=-
      --server=/cluster.local/127.0.0.1#10053
      --server=/in-addr.arpa/127.0.0.1#10053
      --server=/ip6.arpa/127.0.0.1#10053
    State:		Running
      Started:		Wed, 12 Jul 2017 17:35:36 +0800
    Last State:		Terminated
      Reason:		Error
      Exit Code:	137
      Started:		Wed, 12 Jul 2017 17:33:26 +0800
      Finished:		Wed, 12 Jul 2017 17:35:35 +0800
    Ready:		True
    Restart Count:	2
    Requests:
      cpu:		150m
      memory:		20Mi
    Liveness:		http-get http://:10054/healthcheck/dnsmasq delay=60s timeout=5s period=10s #success=1 #failure=5
    Environment:	<none>
    Mounts:
      /etc/k8s/dns/dnsmasq-nanny from kube-dns-config (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-dns-token-xhmbk (ro)
  sidecar:
    Container ID:	docker://d68e48e3303b26778758281bfa24f8078a7e4e4ad02c0f674527b838ab53513b
    Image:		gcr.io/google_containers/k8s-dns-sidecar-amd64:1.14.4
    Image ID:		docker-pullable://gcr.io/google_containers/k8s-dns-sidecar-amd64@sha256:97074c951046e37d3cbb98b82ae85ed15704a290cce66a8314e7f846404edde9
    Port:		10054/TCP
    Args:
      --v=2
      --logtostderr
      --probe=kubedns,127.0.0.1:10053,kubernetes.default.svc.cluster.local,5,A
      --probe=dnsmasq,127.0.0.1:53,kubernetes.default.svc.cluster.local,5,A
    State:		Running
      Started:		Wed, 12 Jul 2017 17:35:36 +0800
    Last State:		Terminated
      Reason:		Error
      Exit Code:	2
      Started:		Wed, 12 Jul 2017 17:33:26 +0800
      Finished:		Wed, 12 Jul 2017 17:35:05 +0800
    Ready:		True
    Restart Count:	2
    Requests:
      cpu:		10m
      memory:		20Mi
    Liveness:		http-get http://:10054/metrics delay=60s timeout=5s period=10s #success=1 #failure=5
    Environment:	<none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-dns-token-xhmbk (ro)
Conditions:
  Type		Status
  Initialized 	True 
  Ready 	False 
  PodScheduled 	True 
Volumes:
  kube-dns-config:
    Type:	ConfigMap (a volume populated by a ConfigMap)
    Name:	kube-dns
    Optional:	true
  kube-dns-token-xhmbk:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	kube-dns-token-xhmbk
    Optional:	false
QoS Class:	Burstable
Node-Selectors:	<none>
Tolerations:	CriticalAddonsOnly
		node-role.kubernetes.io/master:NoSchedule
		node.alpha.kubernetes.io/notReady:NoExecute for 300s
		node.alpha.kubernetes.io/unreachable:NoExecute for 300s
Events:
  FirstSeen	LastSeen	Count	From					SubObjectPath			Type		Reason			Message
  ---------	--------	-----	----					-------------			--------	------			-------
  13m		5m		32	default-scheduler							Warning		FailedScheduling	no nodes available to schedule pods
  5m		5m		1	default-scheduler							Normal		Scheduled		Successfully assigned kube-dns-2425271678-l66cq to m03v
  5m		5m		1	kubelet, m03v					Normal		SuccessfulMountVolume	MountVolume.SetUp succeeded for volume "kube-dns-config" 
  5m		5m		1	kubelet, m03v					Normal		SuccessfulMountVolume	MountVolume.SetUp succeeded for volume "kube-dns-token-xhmbk" 
  3m		2m		2	kubelet, m03v	spec.containers{kubedns}	Warning		Unhealthy		Liveness probe failed: Get http://192.168.54.0:10054/healthcheck/kubedns: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
  5m		1m		4	kubelet, m03v	spec.containers{kubedns}	Normal		Started			Started container
  5m		1m		4	kubelet, m03v	spec.containers{kubedns}	Normal		Created			Created container
  5m		1m		4	kubelet, m03v	spec.containers{kubedns}	Normal		Pulled			Container image "gcr.io/google_containers/k8s-dns-kube-dns-amd64:1.14.4" already present on machine
  4m		1m		9	kubelet, m03v	spec.containers{dnsmasq}	Warning		Unhealthy		Liveness probe failed: Get http://192.168.54.0:10054/healthcheck/dnsmasq: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
  3m		1m		2	kubelet, m03v	spec.containers{sidecar}	Normal		Killing			Killing container with id docker://sidecar:pod "kube-dns-2425271678-l66cq_kube-system(a770cda1-66e3-11e7-900d-00505682fcc1)" container "sidecar" is unhealthy, it will be killed and re-created.
  4m		1m		9	kubelet, m03v	spec.containers{sidecar}	Warning		Unhealthy		Liveness probe failed: Get http://192.168.54.0:10054/metrics: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
  2m		42s		2	kubelet, m03v	spec.containers{dnsmasq}	Normal		Killing			Killing container with id docker://dnsmasq:pod "kube-dns-2425271678-l66cq_kube-system(a770cda1-66e3-11e7-900d-00505682fcc1)" container "dnsmasq" is unhealthy, it will be killed and re-created.
  5m		42s		3	kubelet, m03v	spec.containers{sidecar}	Normal		Pulled			Container image "gcr.io/google_containers/k8s-dns-sidecar-amd64:1.14.4" already present on machine
  5m		41s		3	kubelet, m03v	spec.containers{sidecar}	Normal		Started			Started container
  5m		41s		3	kubelet, m03v	spec.containers{sidecar}	Normal		Created			Created container
  5m		41s		3	kubelet, m03v	spec.containers{dnsmasq}	Normal		Started			Started container
  5m		41s		3	kubelet, m03v	spec.containers{dnsmasq}	Normal		Created			Created container
  5m		41s		3	kubelet, m03v	spec.containers{dnsmasq}	Normal		Pulled			Container image "gcr.io/google_containers/k8s-dns-dnsmasq-nanny-amd64:1.14.4" already present on machine
  5m		22s		25	kubelet, m03v	spec.containers{kubedns}	Warning		Unhealthy		Readiness probe failed: Get http://192.168.54.0:8081/readiness: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
  1m		1s		6	kubelet, m03v	spec.containers{kubedns}	Warning		BackOff			Back-off restarting failed container
  1m		1s		6	kubelet, m03v					Warning		FailedSync		Error syncing pod
```

```
[root@m01v ~]# kubectl logs kube-dns-2425271678-l66cq kubedns -n kube-system 
I0712 09:36:42.452214       1 dns.go:48] version: 1.14.3-4-gee838f6
I0712 09:36:42.452893       1 server.go:70] Using configuration read from directory: /kube-dns-config with period 10s
I0712 09:36:42.452932       1 server.go:113] FLAG: --alsologtostderr="false"
I0712 09:36:42.452946       1 server.go:113] FLAG: --config-dir="/kube-dns-config"
I0712 09:36:42.452951       1 server.go:113] FLAG: --config-map=""
I0712 09:36:42.452954       1 server.go:113] FLAG: --config-map-namespace="kube-system"
I0712 09:36:42.452957       1 server.go:113] FLAG: --config-period="10s"
I0712 09:36:42.452961       1 server.go:113] FLAG: --dns-bind-address="0.0.0.0"
I0712 09:36:42.452963       1 server.go:113] FLAG: --dns-port="10053"
I0712 09:36:42.452968       1 server.go:113] FLAG: --domain="cluster.local."
I0712 09:36:42.452972       1 server.go:113] FLAG: --federations=""
I0712 09:36:42.452976       1 server.go:113] FLAG: --healthz-port="8081"
I0712 09:36:42.452979       1 server.go:113] FLAG: --initial-sync-timeout="1m0s"
I0712 09:36:42.452982       1 server.go:113] FLAG: --kube-master-url=""
I0712 09:36:42.452985       1 server.go:113] FLAG: --kubecfg-file=""
I0712 09:36:42.452988       1 server.go:113] FLAG: --log-backtrace-at=":0"
I0712 09:36:42.452993       1 server.go:113] FLAG: --log-dir=""
I0712 09:36:42.452996       1 server.go:113] FLAG: --log-flush-frequency="5s"
I0712 09:36:42.452998       1 server.go:113] FLAG: --logtostderr="true"
I0712 09:36:42.453001       1 server.go:113] FLAG: --nameservers=""
I0712 09:36:42.453003       1 server.go:113] FLAG: --stderrthreshold="2"
I0712 09:36:42.453006       1 server.go:113] FLAG: --v="2"
I0712 09:36:42.453009       1 server.go:113] FLAG: --version="false"
I0712 09:36:42.453013       1 server.go:113] FLAG: --vmodule=""
I0712 09:36:42.453127       1 server.go:176] Starting SkyDNS server (0.0.0.0:10053)
I0712 09:36:42.453423       1 server.go:198] Skydns metrics enabled (/metrics:10055)
I0712 09:36:42.453449       1 dns.go:147] Starting endpointsController
I0712 09:36:42.453458       1 dns.go:150] Starting serviceController
I0712 09:36:42.453705       1 logs.go:41] skydns: ready for queries on cluster.local. for tcp://0.0.0.0:10053 [rcache 0]
I0712 09:36:42.453714       1 logs.go:41] skydns: ready for queries on cluster.local. for udp://0.0.0.0:10053 [rcache 0]
I0712 09:36:42.953653       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:43.453653       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:43.953666       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:44.453672       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:44.953639       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:45.453650       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:45.953650       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:46.453663       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:46.953665       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:47.453628       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:47.953677       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:48.453663       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:48.953661       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:49.453655       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:49.953650       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:50.453659       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:50.953662       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:51.453652       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:51.953654       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:52.453632       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:52.953643       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:53.453657       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:53.953648       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:54.453660       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:54.953643       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:55.453651       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:55.953649       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:56.453644       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:56.953664       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:57.453622       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:57.953656       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:58.453656       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:58.953686       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:59.453657       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:36:59.953649       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:00.453662       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:00.953668       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:01.453658       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:01.953658       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:02.453629       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:02.953653       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:03.453653       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:03.953661       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:04.453664       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:04.953630       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:05.453650       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:05.953650       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:06.453658       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:06.953657       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:07.453621       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:07.953660       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:08.453656       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:08.953654       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:09.453679       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:09.953641       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:10.453650       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:10.953661       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:11.453667       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:11.953661       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:12.453626       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
E0712 09:37:12.454204       1 reflector.go:199] k8s.io/dns/vendor/k8s.io/client-go/tools/cache/reflector.go:94: Failed to list *v1.Endpoints: Get https://10.96.0.1:443/api/v1/endpoints?resourceVersion=0: dial tcp 10.96.0.1:443: i/o timeout
E0712 09:37:12.454254       1 reflector.go:199] k8s.io/dns/vendor/k8s.io/client-go/tools/cache/reflector.go:94: Failed to list *v1.Service: Get https://10.96.0.1:443/api/v1/services?resourceVersion=0: dial tcp 10.96.0.1:443: i/o timeout
I0712 09:37:12.953657       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:13.453653       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:13.953664       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:14.453664       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:14.953642       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:15.453653       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:15.953674       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:16.453670       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:16.953688       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:17.453624       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:17.953661       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:18.453663       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:18.953673       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:19.453640       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:19.953630       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:20.453644       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:20.953675       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:21.453659       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:21.953665       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:22.453634       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:22.953650       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:23.453656       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:23.953674       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:24.453663       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:24.953659       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:25.453649       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:25.953674       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:26.453663       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:26.953669       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:27.453640       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:27.953650       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
I0712 09:37:28.453655       1 dns.go:174] Waiting for services and endpoints to be initialized from apiserver...
```     

```
[root@m01v ~]# kubectl logs kube-dns-2425271678-l66cq dnsmasq -n kube-system 
I0712 09:37:46.424992       1 main.go:76] opts: {{/usr/sbin/dnsmasq [-k --cache-size=1000 --log-facility=- --server=/cluster.local/127.0.0.1#10053 --server=/in-addr.arpa/127.0.0.1#10053 --server=/ip6.arpa/127.0.0.1#10053] true} /etc/k8s/dns/dnsmasq-nanny 10000000000}
I0712 09:37:46.425170       1 nanny.go:86] Starting dnsmasq [-k --cache-size=1000 --log-facility=- --server=/cluster.local/127.0.0.1#10053 --server=/in-addr.arpa/127.0.0.1#10053 --server=/ip6.arpa/127.0.0.1#10053]
I0712 09:37:46.544187       1 nanny.go:111] 
W0712 09:37:46.544213       1 nanny.go:112] Got EOF from stdout
I0712 09:37:46.544236       1 nanny.go:108] dnsmasq[13]: started, version 2.76 cachesize 1000
I0712 09:37:46.544246       1 nanny.go:108] dnsmasq[13]: compile time options: IPv6 GNU-getopt no-DBus no-i18n no-IDN DHCP DHCPv6 no-Lua TFTP no-conntrack ipset auth no-DNSSEC loop-detect inotify
I0712 09:37:46.544250       1 nanny.go:108] dnsmasq[13]: using nameserver 127.0.0.1#10053 for domain ip6.arpa 
I0712 09:37:46.544253       1 nanny.go:108] dnsmasq[13]: using nameserver 127.0.0.1#10053 for domain in-addr.arpa 
I0712 09:37:46.544256       1 nanny.go:108] dnsmasq[13]: using nameserver 127.0.0.1#10053 for domain cluster.local 
I0712 09:37:46.544260       1 nanny.go:108] dnsmasq[13]: reading /etc/resolv.conf
I0712 09:37:46.544263       1 nanny.go:108] dnsmasq[13]: using nameserver 127.0.0.1#10053 for domain ip6.arpa 
I0712 09:37:46.544266       1 nanny.go:108] dnsmasq[13]: using nameserver 127.0.0.1#10053 for domain in-addr.arpa 
I0712 09:37:46.544269       1 nanny.go:108] dnsmasq[13]: using nameserver 127.0.0.1#10053 for domain cluster.local 
I0712 09:37:46.544272       1 nanny.go:108] dnsmasq[13]: using nameserver 10.20.11.10#53
I0712 09:37:46.544277       1 nanny.go:108] dnsmasq[13]: read /etc/hosts - 7 addresses
```

```
[root@m01v ~]# kubectl logs kube-dns-2425271678-l66cq sidecar -n kube-system 
ERROR: logging before flag.Parse: I0712 09:37:47.202881       1 main.go:48] Version v1.14.3-4-gee838f6
ERROR: logging before flag.Parse: I0712 09:37:47.204732       1 server.go:45] Starting server (options {DnsMasqPort:53 DnsMasqAddr:127.0.0.1 DnsMasqPollIntervalMs:5000 Probes:[{Label:kubedns Server:127.0.0.1:10053 Name:kubernetes.default.svc.cluster.local. Interval:5s Type:1} {Label:dnsmasq Server:127.0.0.1:53 Name:kubernetes.default.svc.cluster.local. Interval:5s Type:1}] PrometheusAddr:0.0.0.0 PrometheusPort:10054 PrometheusPath:/metrics PrometheusNamespace:kubedns})
ERROR: logging before flag.Parse: I0712 09:37:47.204784       1 dnsprobe.go:75] Starting dnsProbe {Label:kubedns Server:127.0.0.1:10053 Name:kubernetes.default.svc.cluster.local. Interval:5s Type:1}
ERROR: logging before flag.Parse: I0712 09:37:47.204844       1 dnsprobe.go:75] Starting dnsProbe {Label:dnsmasq Server:127.0.0.1:53 Name:kubernetes.default.svc.cluster.local. Interval:5s Type:1}

```

**Anything else we need to know?**:
Looking up for similar issue. and found #27671. But I'm not sure whether it is the same issue.

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.0", GitCommit:"d3ada0119e776222f11ec7945e6d860061339aad", GitTreeState:"clean", BuildDate:"2017-06-29T23:15:59Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.0", GitCommit:"d3ada0119e776222f11ec7945e6d860061339aad", GitTreeState:"clean", BuildDate:"2017-06-29T22:55:19Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```

- OS (e.g. from /etc/os-release):
```
NAME="Red Hat Enterprise Linux Server"
VERSION="7.2 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="7.2"
PRETTY_NAME="Red Hat Enterprise Linux Server 7.2 (Maipo)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:7.2:GA:server"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 7"
REDHAT_BUGZILLA_PRODUCT_VERSION=7.2
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="7.2"
```
- Kernel (e.g. `uname -a`):
``Linux 3.10.0-327.el7.x86_64 #1 SMP Thu Oct 29 17:29:29 EDT 2015 x86_64 x86_64 x86_64 GNU/Linux``
- Install tools:
- Others:

**Comment by zhangqx2010** [comment]
After delete calico, kube-dns come back.
It looks likely due to the calico yaml not compatible with kube 1.7.
Trying to figure out the issue.

**Comment by bowei** [comment]
cc: @dnardo

**Comment by bowei** [comment]
@dnardo -- nevermind, this isn't related to network policy

**Comment by zhangqx2010** [comment]
Find something that may cause this problem.

1. I need to manually configure one static ip for cbr. (following: https://kubernetes.io/docs/getting-started-guides/scratch/#networking)
2. Calico will use the old ip address for new deployment if calico has been deployed earlier. Even when I delete the cache /var/etcd/calico-data/ , which I think is really strange. And I also investigated the etcd content for k8s, no KV is found for calico.

After I set up one cbr on each node and redeployed calico, the cluster works fine now
